### ISSUE TITLE
null_resource asks for 'content' or 'source'

### ISSUE THREAD

**Issue by shantanugadgil** [hashicorp/terraform#15186]
### Terraform Version
Terraform v0.9.8

### Affected Resource(s)
Please list the resources as a list, for example:
- null_resource

I have a `null_resource` which has a few triggers and a bunch of `file` provisioners.

Things were working fine in `v0.9.6`
Now with `v0.9.8` when I do `plan`, it complains:

```
- * null_resource.data_dir: Must provide one of 'content' or 'source'
```

Seems to be something similar to
- GH-15177

Regards,
Shantanu

**Comment by shantanugadgil** [comment]
Switching my binary back to v0.9.6 has made it work.

**Comment by onorua** [comment]
Duplicate of https://github.com/hashicorp/terraform/issues/15177

**Comment by shantanugadgil** [comment]
Hi,

After looking at @onorua  's comment, I closely examined my configuration.
I thought it was something to do with the `null_resource`

Actually, the problem is:
```hcl
  provisioner "file" {
    content = "${data.template_file.test.rendered}"
    destination = "foo"
  }
```

Changing it to the following makes the `plan` warning go away, but doesn't mean the same, so the problem remain (overall)
```hcl
  provisioner "file" {
    content = "boo"
    destination = "foo"
  }
```

For my use case, I have variables substituted in the template_file.

Regards,
Shantanu

**Comment by shantanugadgil** [comment]
Ok, so I have changed by "data template_file" to be used directly in the "content" block of the file provisioner, but this doesn't detect if I make changes in the "content =" line.

I had previously set a trigger on the sha256 of the template file which I was using, so this breaks my current workflow. 😢  😞 

Hope this gets fixed soon,

Regards,
Shantanu

**Comment by svanharmelen** [comment]
This is fixed in 0.9.9 :wink:
